sched/scheduler/scheduler.go

```diff
@@ -8,6 +10,10 @@ import (
 	"sync"
 )
 
+type Scheduler interface {
```

Comment:
Either way we want to separate out the management of dead letter queues. So either Scheduler calls that piece or that piece calls Scheduler. It's the latter right now and that seems fine, although I'd probably want an interface/struct instead of a standalone function.